### PR TITLE
Fix (quite) all compilation warnings.

### PR DESCRIPTION
**Pull request by adjamhub**
To be fair, there is one missing... about glib g_slist_free_full.
But I had to give up ;)

### CODE REVIEW

File: src/filedialoghelper.cpp [hunk 1]
```diff
@@ -407,9 +413,17 @@ void FileDialogHelper::saveSettings() {
...
-        settings.setValue(QStringLiteral("HiddenPlaces"), sl);
+    else {
+#if (QT_VERSION >= QT_VERSION_CHECK(5,14,0))
+        QStringList hiddenPlacesList = settings.value(QStringLiteral("HiddenPlaces")).toStringList();
+        QSet<QString> hiddenPlacesSet = QSet<QString>(hiddenPlacesList.begin(), hiddenPlacesList.end());
+#else
+        QSet<QString> hiddenPlacesSet = settings.value(QStringLiteral("HiddenPlaces")).toStringList().toSet();
+#endif
+        if (hiddenPlaces != hiddenPlacesSet) {
+            QStringList sl = hiddenPlaces.values();
```
Comment (tsujan) [inline src/filedialoghelper.cpp:424]: `QSet::toList()` and `QSet::values()` are the same. If you want to use range constructors with Qt ≥ 5.14, you could write:

```
            QStringList sl (hiddenPlaces.begin(), hiddenPlaces.end());
```

File: src/folderview.h [hunk 2]
```diff
@@ -130,7 +130,7 @@ class LIBFM_QT_API FolderView : public QWidget {
     void setCustomColumnWidths(const QList<int> &widths);
 
     QList<int> getHiddenColumns() const {
-        return hiddenColumns_.toList();
+        return hiddenColumns_.values();
```
Comment (tsujan) [inline src/folderview.h:133]: The same as above.

Comment (adjamhub) [reply src/folderview.h:133]: Ok. It's just that the compiler suggested me to use values :)
I'll ifdef it then

**Review by tsujan — APPROVED** [review]
Everything seems OK. Thanks!

**Pull request merged**